File: API/Entities/Attendance.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities
{
    [Table("Attendances")]
    public class Attendance : BaseEntity
    {
        // [Key]
        // public int AttendanceId { get; set; }

        [Required]
        public int StudentId { get; set; }

        [Required]
        public int ScheduleId { get; set; }

        [Required]
        public int QRCodeId { get; set; }

        [Required]
        public DateTime AttendanceDate { get; set; }

        [Required]
        public bool Status { get; set; }  // true for Present, false for Absent

        [StringLength(500)]
        public string? StudentNote { get; set; }

        [StringLength(500)]
        public string? DoctorNote { get; set; }

        // Navigation Properties
        [ForeignKey("StudentId")]
        public Student Student { get; set; }

        [ForeignKey("ScheduleId")]
        public LectureSchedule LectureSchedule { get; set; }

        [ForeignKey("QRCodeId")]
        public QRCode QRCode { get; set; }
    }
}

File: API/Flutter App/coordinator_dashpord_for_studentmark/lib/core/models/attendance.dart
import 'package:coordinator_dashpord_for_studentmark/core/models/lecture_schedule.dart';
import 'package:coordinator_dashpord_for_studentmark/core/models/student.dart';
import 'qr_code.dart';

class Attendance {
  final int id;
  final int studentId;
  final int scheduleId;
  final int qrCodeId;
  final DateTime attendanceDate;
  final bool status;
  final String? studentNote;
  final String? doctorNote;
  // final DateTime createdAt;
  // final DateTime? updatedAt;

  // Navigation properties
  final Student? student;
  final LectureSchedule? lectureSchedule;
  final QRCode? qrCode;

  Attendance({
    required this.id,
    required this.studentId,
    required this.scheduleId,
    required this.qrCodeId,
    required this.attendanceDate,
    required this.status,
    this.studentNote,
    this.doctorNote,
    // required this.createdAt,
    // this.updatedAt,
    this.student,
    this.lectureSchedule,
    this.qrCode,
  });

  factory Attendance.fromJson(Map<String, dynamic> json) {
    return Attendance(
      id: json['id'] as int,
      studentId: json['studentId'] as int,
      scheduleId: json['scheduleId'] as int,
      qrCodeId: json['qrCodeId'] as int,
      attendanceDate: DateTime.parse(json['attendanceDate'] as String),
      status: json['status'] as bool,
      studentNote: json['studentNote'] as String?,
      doctorNote: json['doctorNote'] as String?,
      // createdAt: DateTime.parse(json['createdAt'] as String),
      // updatedAt: json['updatedAt'] != null
      //     ? DateTime.parse(json['updatedAt'] as String)
      //     : null,
      student:
          json['student'] != null ? Student.fromJson(json['student']) : null,
      lectureSchedule: json['lectureSchedule'] != null
          ? LectureSchedule.fromJson(json['lectureSchedule'])
          : null,
      qrCode: json['qrCode'] != null ? QRCode.fromJson(json['qrCode']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'studentId': studentId,
      'scheduleId': scheduleId,
      'qrCodeId': qrCodeId,
      'attendanceDate': attendanceDate.toIso8601String(),
      'status': status,
      'studentNote': studentNote,
      'doctorNote': doctorNote,
      // 'createdAt': createdAt.toIso8601String(),
      // 'updatedAt': updatedAt?.toIso8601String(),
      'student': student?.toJson(),
      'lectureSchedule': lectureSchedule?.toJson(),
      'qrCode': qrCode?.toJson(),
    };
  }
}
